client/asset/dcr/dcr.go

```diff
@@ -1483,86 +1496,382 @@ func (dcr *ExchangeWallet) SignMessage(coin asset.Coin, msg dex.Bytes) (pubkeys,
+}
+
+func (dcr *ExchangeWallet) lookupTxOutput(ctx context.Context, op outPoint, mempool bool, pkScript []byte, earliestTxTime time.Time) (*txOutput, outputSpendStatus, error) {
+	errorOut := func(err error) (*txOutput, outputSpendStatus, error) {
+		return nil, outputSpendStatusUnknown, err
+	}
+
+	// First perform wallet lookup, may be able to find the output
+	// even if it doesn't pay to the wallet.
+	txOut, spendStatus, err := dcr.checkWalletForTxOutput(ctx, op, mempool)
+	if err != nil && err != asset.CoinNotFoundError {
+		// Do not return CoinNotFoundError yet, check cache first.
+		return errorOut(err)
+	}
+
+	if txOut != nil {
+		// Output is found in wallet. Is the spend status known?
+		if spendStatus != outputSpendStatusUnknown {
+			dcr.log.Debugf("Found output %s in wallet, spent %t", op, spendStatus == outputSpendStatusSpent)
+			return txOut, spendStatus, nil
+		} else if len(pkScript) == 0 {
+			dcr.log.Debugf("Found output %s in wallet, but spend status cannot be determined", op)
+			return txOut, spendStatus, nil
+		}
+	}
+
+	// If the output was not found by the wallet, check the externalTx
+	// cache and if also not found, scan block filters to find the output.
+	var eTxOut *externalTxOutput
+	if txOut == nil {
+		if len(pkScript) == 0 {
+			dcr.log.Debugf("Output %s NOT found in wallet, checking external tx cache", op)
+		} else {
+			dcr.log.Debugf("Output %s NOT found in wallet, checking cache/block filters", op)
+		}
+		eTxOut, err = dcr.lookupTxOutWithBlockFilters(ctx, op, pkScript, earliestTxTime)
+		if err != nil {
+			return errorOut(err)
+		}
+	} else {
+		dcr.log.Debugf("Found output %s in wallet, will determine spend status using block filters", op)
+		// TODO: If it is found in wallet, just assume unspent?
+		eTxOut = &externalTxOutput{
+			txOutput: txOut,
+		}
 	}
-	return &asset.AuditInfo{
-		Coin:       newOutput(txHash, vout, value, txTree),
-		Contract:   contract,
-		SecretHash: secretHash,
-		Recipient:  receiver.String(),
-		Expiration: time.Unix(int64(stamp), 0).UTC(),
-	}, nil
+
+	// If we have the output details from the wallet lookup or
+	// the block filters scan above, let's check if it is spent.
+	spendStatus, err = dcr.checkOutputSpendStatus(ctx, eTxOut)
+	return eTxOut.txOutput, spendStatus, err
+}
+
+// checkWalletForTxOutput attempts to find and return details for the specified
+// tx output. Returns asset.CoinNotFoundError if the output is not found.
+// NOTE: SPV wallets are only able to lookup outputs for transactions that are
+// tracked by the wallet.
+func (dcr *ExchangeWallet) checkWalletForTxOutput(ctx context.Context, op outPoint, mempool bool) (*txOutput, outputSpendStatus, error) {
+	errorOut := func(err error) (*txOutput, outputSpendStatus, error) {
+		return nil, outputSpendStatusUnknown, err
+	}
+
+	// First use wallet.GetTxOut to look up the output.
+	unspentTxOut, txTree, err := dcr.wallet.GetTxOut(ctx, &op.txHash, op.vout, wire.TxTreeUnknown, mempool)
```

Comment:
Thought I renamed `GetTxOut` to `UnspentOutput` since it really only returns tx outputs that are unspent (hence the references to UnspentOutput you noted in some docs). I will complete that renaming to make things less confusing, since `gettxout` is quite really an rpc-only concept for fetching unspent outputs.

`checkWalletForTxOutput` does check wallet transactions but first calls the `wallet.GetTxOut` method to attempt finding the (unspent) output _using the wallet_. `GetTxOut` is a method the wallet provides for looking up unspent outputs. The difference is that full node wallets are able to consult the connected full node to find unspent outputs that may not be related to the wallet, while spv wallets only check unspent wallet outputs.